### PR TITLE
fix(goal_planner): fix goal search for narrow shoulder lane

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Fix goal search for narrow shoulder lane considering patterns when the bound of the vehicle at pull over lane's center line is outside the bound of the lanelet.

before (skip checking is in lane)
 
image

after

image

image

image

image

## Tests performed

psim

https://evaluation.tier4.jp/evaluation/reports/25c84c50-d734-5075-9ab2-fc1e45884ff6?project_id=prd_jt

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/utils/utils.hpp [hunk 1]
```diff
@@ -323,8 +323,8 @@ PathPointWithLaneId insertStopPoint(const double length, PathWithLaneId & path);
 double getSignedDistanceFromBoundary(
   const lanelet::ConstLanelets & shoulder_lanelets, const Pose & pose, const bool left_side);
 std::optional<double> getSignedDistanceFromBoundary(
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/utils.hpp:325]: Is it also an option to take the `vehicle_info_util::VehicleInfo` type as an argument?
This member variable is necessary for safety check in goal_planner, so i added in this [PR](https://github.com/autowarefoundation/autoware.universe/pull/4859)

Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/utils.hpp:325]: yes it is also ok for me.

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/utils.hpp:325]: merged in the [PR](https://github.com/autowarefoundation/autoware.universe/pull/4859)

**Comment by kyoichi-sugahara** [comment]
@kosuke55 
could you give me time to check this PR?
If it's urgent I will take a look briefly

**Comment by kosuke55** [comment]
@kyoichi-sugahara 
OK, thanks!!b

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

**Pull request merged**